nixos/modules/system/boot/kernel.nix

```diff
@@ -187,7 +187,7 @@ in
 
   ###### implementation
 
-  config = mkIf (!config.boot.isContainer) {
+  config = mkIf config.boot.initrd.enable {
```

Comment:
is there any particular need to disable this config based on `boot.initrd.enable`? Everything related to initrd will be disabled in that case, why disable rest?

For example, `system.build.kernel`, `boot.kernelParams`, `hardware.firmware` are irrelevant to initrd, right?

cc also @acertain to explain usecase.

Reply:
Yeah, it seems we have to separate the container from initrd logic somehow.

Reply:
I've split the the configuration in two sections and used `mkMerge`. I'm not so sure if this is 100% correct.